pkg/bwagreement/testbwagreement/common.go

```diff
@@ -41,3 +65,20 @@ func GenerateRenterBandwidthAllocation(pba *pb.PayerBandwidthAllocation, storage
 	// Combine Signature and Data for RenterBandwidthAllocation
 	return rba, auth.SignMessage(rba, *upID)
 }
+
+// get uplink's public key
+func getUplinkPubKey(uplinkKey crypto.PrivateKey) ([]byte, error) {
```

Comment:
Move this logic into the CreateAgreement

Reply:
What I meant was to pass a crypto.PrivateKey into CreateAgreement.  We'd like the database layer (EG CreateAgreement) to hide details of its implementation (EG that its storing a slice of bytes).

Reply:
am not sure passing uplink's private key as an argument would be a good idea. This file is used only for test cases.

Reply:
doh, I meant public key

Reply:
I think you can use `identity.Leaf.PublicKey.(*ecdsa.PublicKey)` to get the public key thought the private key, then move the conversion to []byte into the database access layer.